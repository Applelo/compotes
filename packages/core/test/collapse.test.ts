import { afterAll, beforeAll, describe, expect, it } from 'vitest'
import type { Browser } from 'playwright'
import { chromium } from 'playwright'

let browser: Browser

beforeAll(async () => {
  browser = await chromium.launch()
})

afterAll(async () => {
  browser.close()
})

describe('collapse', async () => {
  it.concurrent('events', async () => {
    const page = await browser.newPage()
    const events = [
      'c.collapse.show',
      'c.collapse.hide',
      'c.collapse.hidden',
      'c.collapse.shown',
    ]
    const showEvent = page.waitForEvent('console', msg => msg.text().includes('c.collapse.show'))
    const hideEvent = page.waitForEvent('console', msg => msg.text().includes('c.collapse.hide'))
    const shownEvent = page.waitForEvent('console', msg => msg.text().includes('c.collapse.shown'))
    const hiddenEvent = page.waitForEvent('console', msg => msg.text().includes('c.collapse.hidden'))
    await page.goto('http://localhost:3000/collapse.html')

    await page.evaluate(() => {
      const el = document.querySelector('.c-collapse')
      if (!el)
        return
      [
        'c.collapse.show',
        'c.collapse.hide',
        'c.collapse.hidden',
        'c.collapse.shown',
      ].forEach((e) => {
        el.addEventListener(e, () => {
          // eslint-disable-next-line no-console
          console.log(e)
        })
      })
    })

    const triggerEl = page.locator('.c-collapse-trigger').first()
    await triggerEl.click()
    await shownEvent
    await triggerEl.click()

    const result = (await Promise.all([
      showEvent,
      shownEvent,
      hideEvent,
      hiddenEvent,
    ])).map(item => item.text())

    events.forEach(e => expect(result).toContain(e))
  })
})
