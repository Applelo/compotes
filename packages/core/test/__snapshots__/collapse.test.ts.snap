// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`collapse > destroy 1`] = `
"<!DOCTYPE html><html lang="en"><head>
    <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>🍯</text></svg>">
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Collapse</title>
    <script type="module" crossorigin="" src="/assets/collapse-DFX_hf7s.js"></script>
    <link rel="modulepreload" crossorigin="" href="/assets/_parent-CjRBGSFb.js">
    <link rel="modulepreload" crossorigin="" href="/assets/animation-CH8HhKvA.js">
    <link rel="stylesheet" crossorigin="" href="/assets/marquee-CjakcbCe.css">
  </head>
  <body>
    <div>
      <button class="c-collapse-trigger" aria-controls="accordion-1">
        Accordion 2
      </button>
      <div class="c-collapse" id="accordion-1" style="transition: height .2s;">
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Mollitia facere possimus impedit facilis culpa illo earum deserunt consequuntur minus. Ad et qui labore reprehenderit magnam exercitationem placeat magni nesciunt suscipit.
        </p>
      </div>
    </div>
    <button class="js-destroy">
      Destroy
    </button>

  

</body></html>"
`;
